File: Data Structures & Algorithms/anagram-groups/submission-0.cpp

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> buckets;   // key = sorted word

        for (const string& s : strs) {
            string key = s;
            sort(key.begin(), key.end());               // O(L log L)
            buckets[key].push_back(s);
        }

        vector<vector<string>> result;
        for (auto& [_, group] : buckets) {               // C++17 structured binding
            result.push_back(move(group));
        }
        return result;
    }
};

File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        st = "".join(sorted(s))
        tt = "".join(sorted(t))
        if st!=tt:
            return False
        return True
        